### PR TITLE
BADGERS-350: upgrade Actions workflows to use node20 & latest action versions

### PR DESCRIPTION
**Pull request by tom-coward**
📺 What

Upgrades all Actions workflows to use Node 20 (upgrading from 16), and all actions to their latest versions.

🛠 How

Define `node-version` as 20 in `setup-node` steps.
Define versions of all actions to the current latest.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -39,11 +39,11 @@ jobs:
...
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       # Initializes the CodeQL tools for scanning.
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@v1
+        uses: github/codeql-action/init@v3
         with:
           languages: ${{ matrix.language }}
```
Comment (ShiningTrapez) [inline .github/workflows/codeql-analysis.yml:48]: You could inline this Matrix in the same way you did for the Node Versions, just to be consistent

**Review by ShiningTrapez — DISMISSED** [review]
LGTM!

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (tom-coward) [reply .github/workflows/codeql-analysis.yml:48]: Good spot - have done this

**Review by ShiningTrapez — DISMISSED** [review]

**Review by ShiningTrapez — APPROVED** [review]

**Comment by tom-coward** [comment]
Bypassing checks to get this merged as codeql can't find an old codeql analysis run associated with the PR to compare to after switching out the matrix languages to be hardcoded - we think this should be fine for future PRs but will verify this after merging:
image
image

**Pull request merged**